File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/C++/s863488345_commented.cpp
#include <iostream>
#include <vector>
#include <string>
using namespace std;

int mapNum;
int h, w;
int bomberActNum;
int xOfTank, yOfTank;
int xOfBomber, yOfBomber;
int s, k, a, b;
int x, y;
vector<vector<char>> battleTown;

void moveBomber(int a, int b) {
    if (0 <= a && a < h && 0 <= b && b < w) {
        if (battleTown[a][b] == '^') {
            for (int i = a - 1; i >= 0; i--) {
                if (a <= 0) {
                    return;
                }
                if (battleTown[a - 1][b] == '*') {
                    battleTown[a - 1][b] = '.';
                    return;
                } else if (battleTown[a - 1][b] == '#') {
                    return;
                } else {
                    a--;
                }
            }
        } else if (battleTown[a][b] == 'v') {
            for (int i = a + 1; i < h; i++) {
                if (battleTown[a + 1][b] == '*') {
                    battleTown[a + 1][b] = '.';
                    return;
                } else if (battleTown[a + 1][b] == '#') {
                    return;
                } else {
                    a++;
                }
            }
        } else if (battleTown[a][b] == '<') {
            for (int i = b - 1; i >= 0; i--) {
                if (battleTown[a][b - 1] == '*') {
                    battleTown[a][b - 1] = '.';
                    return;
                } else if (battleTown[a][b - 1] == '#') {
                    return;
                } else {
                    b--;
                }
            }
        } else if (battleTown[a][b] == '>') {
            for (int i = b + 1; i < w; i++) {
                if (battleTown[a][b + 1] == '*') {
                    battleTown[a][b + 1] = '.';
                    return;
                } else if (battleTown[a][b + 1] == '#') {
                    return;
                } else {
                    b++;
                }
            }
        }
    }
}

int main() {
    cin >> mapNum;
    for (int i = 0; i < mapNum; i++) {
        if (i > 0) {
            cout << endl;
        }
        cin >> h >> w;
        battleTown = vector<vector<char>>(h, vector<char>(w));
        x = -1;
        y = -1;
        for (int j = 0; j < h; j++) {
            string s;
            cin >> s;
            for (int k = 0; k < w; k++) {
                battleTown[j][k] = s[k];
                if (battleTown[j][k] == '^' || battleTown[j][k] == 'v' || 
                    battleTown[j][k] == '<' || battleTown[j][k] == '>') {
                    x = j;
                    y = k;
                }
            }
        }
        cin >> bomberActNum;
        string act;
        cin >> act;
        for (int l = 0; l < bomberActNum; l++) {
            char c = act[l];
            if (c == 'U') {
                battleTown[x][y] = '^';
                if (x - 1 >= 0 && battleTown[x - 1][y] == '.') {
                    battleTown[x - 1][y] = '^';
                    battleTown[x][y] = '.';
                    x--;
                }
            } else if (c == 'D') {
                battleTown[x][y] = 'v';
                if (x + 1 < h && battleTown[x + 1][y] == '.') {
                    battleTown[x + 1][y] = 'v';
                    battleTown[x][y] = '.';
                    x++;
                }
            } else if (c == 'L') {
                battleTown[x][y] = '<';
                if (y - 1 >= 0 && battleTown[x][y - 1] == '.') {
                    battleTown[x][y - 1] = '<';
                    battleTown[x][y] = '.';
                    y--;
                }
            } else if (c == 'R') {
                battleTown[x][y] = '>';
                if (y + 1 < w && battleTown[x][y + 1] == '.') {
                    battleTown[x][y + 1] = '>';
                    battleTown[x][y] = '.';
                    y++;
                }
            } else if (c == 'S') {
                a = x;
                b = y;
                moveBomber(a, b);
            }
        }
        for (s = 0; s < h; s++) {
            for (k = 0; k < w; k++) {
                cout << battleTown[s][k];
            }
            cout << endl;
        }
    }
    return 0;
}
// End of Code

File: tasks/translation/output/Qwen/QwQ-32B-AWQ/codenet/direct/comment/uncomment/direct/Java/Python/s863488345_commented.py
import sys

def move_bomber(a, b, battle_town, h, w):
    direction = battle_town[a][b]
    if direction == '^':
        current_a = a
        current_b = b
        while True:
            next_a = current_a - 1
            if next_a < 0:
                break
            cell = battle_town[next_a][current_b]
            if cell == '*':
                battle_town[next_a][current_b] = '.'
                break
            elif cell == '#':
                break
            else:
                current_a = next_a
    elif direction == 'v':
        current_a = a
        current_b = b
        while True:
            next_a = current_a + 1
            if next_a >= h:
                break
            cell = battle_town[next_a][current_b]
            if cell == '*':
                battle_town[next_a][current_b] = '.'
                break
            elif cell == '#':
                break
            else:
                current_a = next_a
    elif direction == '<':
        current_a = a
        current_b = b
        while True:
            next_b = current_b - 1
            if next_b < 0:
                break
            cell = battle_town[current_a][next_b]
            if cell == '*':
                battle_town[current_a][next_b] = '.'
                break
            elif cell == '#':
                break
            else:
                current_b = next_b
    elif direction == '>':
        current_a = a
        current_b = b
        while True:
            next_b = current_b + 1
            if next_b >= w:
                break
            cell = battle_town[current_a][next_b]
            if cell == '*':
                battle_town[current_a][next_b] = '.'
                break
            elif cell == '#':
                break
            else:
                current_b = next_b

def main():
    mapNum = int(sys.stdin.readline())
    for i in range(mapNum):
        if i > 0:
            print()
        h, w = map(int, sys.stdin.readline().split())
        battle_town = []
        x, y = -1, -1
        for row_idx in range(h):
            line = sys.stdin.readline().strip()
            row = list(line)
            battle_town.append(row)
            for col_idx in range(w):
                if row[col_idx] in ['^', 'v', '<', '>']:
                    x = row_idx
                    y = col_idx
        bomberActNum = int(sys.stdin.readline())
        bomberAct = sys.stdin.readline().strip()
        for action in bomberAct:
            if action == 'U':
                new_dir = '^'
                battle_town[x][y] = new_dir
                next_x = x - 1
                next_y = y
                if 0 <= next_x < h and 0 <= next_y < w:
                    if battle_town[next_x][next_y] == '.':
                        battle_town[x][y] = '.'
                        battle_town[next_x][next_y] = new_dir
                        x = next_x
                        y = next_y
            elif action == 'D':
                new_dir = 'v'
                battle_town[x][y] = new_dir
                next_x = x + 1
                next_y = y
                if 0 <= next_x < h and 0 <= next_y < w:
                    if battle_town[next_x][next_y] == '.':
                        battle_town[x][y] = '.'
                        battle_town[next_x][next_y] = new_dir
                        x = next_x
                        y = next_y
            elif action == 'L':
                new_dir = '<'
                battle_town[x][y] = new_dir
                next_x = x
                next_y = y - 1
                if 0 <= next_x < h and 0 <= next_y < w:
                    if battle_town[next_x][next_y] == '.':
                        battle_town[x][y] = '.'
                        battle_town[next_x][next_y] = new_dir
                        x = next_x
                        y = next_y
            elif action == 'R':
                new_dir = '>'
                battle_town[x][y] = new_dir
                next_x = x
                next_y = y + 1
                if 0 <= next_x < h and 0 <= next_y < w:
                    if battle_town[next_x][next_y] == '.':
                        battle_town[x][y] = '.'
                        battle_town[next_x][next_y] = new_dir
                        x = next_x
                        y = next_y
            elif action == 'S':
                move_bomber(x, y, battle_town, h, w)
        for row in battle_town:
            print(''.join(row))

if __name__ == "__main__":
    main() # End of Code
